pkg/docker/containers-view.jsx

```diff
@@ -530,15 +530,19 @@ var ImageList = React.createClass({
     deleteImage: function (image, event) {
         if (event.button !== 0)
             return;
-
-        util.confirm(cockpit.format(_("Delete $0"), image.RepoTags[0]),
-                     _("Are you sure you want to delete this image?"), _("Delete"))
-                .done(function () {
-                    this.props.client.rmi(image.Id)
-                            .fail(function(ex) {
-                                util.show_unexpected_error(ex);
-                            });
+        util.delete_image_confirm(this.props.client, image).done(
+            function(runningContainers, force) {
+                var stopPromises = [];
+                if (runningContainers.length > 0)
+                    $(runningContainers).each(function(index, value) {
+                        stopPromises.push(this.props.client.stop(value));
+                    }.bind(this));
+                cockpit.all(stopPromises).done(function() {
+                    this.props.client.rmi(image.Id, force).fail(function(ex) {
+                        util.show_unexpected_error(ex);
```

Comment:
This code now also exists twice, could it be moved into util as well? Feel free (and in fact encouraged) to rename that to `util.jsx` to use modern syntax there (do that in a  separate commit or PR, in case you have to fix up stuff due to eslint now complaining about syntactic stuff).

I don't quite understand the `force` handling here: If it is false (returned from the confirmation dialog, right?), the above code still stops all running containers -- shouldn't this only happen for `force == true`?

Reply:
I guess it's not super clear. Force remove needs to happen when there are containers (exited or running), running containers need to be stopped. So it's always force remove unless there are no containers depending on the image.

Reply:
Oh, you are saying that if `delete_image_confirm()` succeeds, the user already has agreed to the forcing. Is there actually a case where calling `rmi()` without `force: true` makes any difference? Or is it just being defensive and ensuring we didn't miss to clean up anything else? (which is good, I just want to understand how it works)

Reply:
Correct, force is implied when the user confirms a rmi with depending containers. So you do need the force if there are containers, regardless of state. Force can be false if there are no depending containers.

It's mostly about being clean/meticulous, as we could just pass `force: true` all the time.